wcfsetup/install/files/lib/data/I18nDatabaseObjectList.class.php

```diff
@@ -0,0 +1,54 @@
+<?php
+namespace wcf\data;
+use wcf\system\WCF;
+use wcf\util\StringUtil;
+
+/**
+ * Abstract class for a list of database objects with better sorting of i18n-based columns.
+ *
+ * @author	Florian Gail
+ * @copyright	2001-2017 WoltLab GmbH
+ * @license	GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package	WoltLabSuite\Core\Data
+ * @since	3.1
+ */
+abstract class I18nDatabaseObjectList extends DatabaseObjectList {
+	/**
+	 * Array of fields filled with language items
+	 * this slows down the system, use this only when sorting
+	 * based on an i18n-value.
+	 * The value represents the original field name.
+	 * The key represents the new field name and is optional.
+	 * If no key is provided, the new field will be 'i18n'.$value
+	 * 
+	 * use table.field schema!
+	 * 
+	 * @example   [ 'title' => 'titleSortField' ]
```

Comment:
This example contradicts the explanation above. Should it read `[ 'titleSortField' => 'title' ]`?

Reply:
Yes. But I'll change the behavior since key = original column and value = new column's name should be more logical.

Reply:
@mutec That would allow defining the same field multiple times. Not sure whether that is better. Add an explicit validation then.

Reply:
Better now? I'll clean up the commits if everything is correct.